static/app/views/discover/resultsSearchQueryBuilder.tsx

```diff
@@ -163,11 +180,90 @@ function ResultsSearchQueryBuilder(props: Props) {
   const measurements = useMemo(() => getMeasurements(), []);
   const functionTags = useMemo(() => getFunctionTags(fields), [fields]);
 
+  const featureFlagQuery = useFetchOrganizationTags(
+    {
+      orgSlug: organization.slug,
+      projectIds: projectIdStrings,
+      dataset: AlertsDatasets.ERRORS,
+      useCache: true,
+      useFlagsBackend: true,
+      enabled: includeFeatureFlags,
+      ...dateTimeParams,
+    },
+    {}
+  );
+  const featureFlagTags: TagCollection = useMemo(
+    () =>
+      featureFlagQuery.data?.reduce<TagCollection>((acc, tag) => {
+        // Wrap with flags[""]. flags[] is required for the search endpoint and "" is used to escape special characters.
+        const key = `flags["${tag.key}"]`;
+        acc[key] = {...tag, kind: FieldKind.FEATURE_FLAG, key};
+        return acc;
+      }, {}) || {},
+    [featureFlagQuery.data]
+  );
+
+  const getTagList: TagCollection = useMemo(() => {
+    const measurementsWithKind = getMeasurementTags(measurements, customMeasurements);
+    const orgHasPerformanceView = organization.features.includes('performance-view');
+
+    const combinedTags: TagCollection =
+      dataset === DiscoverDatasets.ERRORS
+        ? Object.assign({}, functionTags, STATIC_FIELD_TAGS_WITHOUT_TRANSACTION_FIELDS)
+        : dataset === DiscoverDatasets.TRANSACTIONS ||
+            dataset === DiscoverDatasets.METRICS_ENHANCED
+          ? Object.assign(
+              {},
+              measurementsWithKind,
+              functionTags,
+              STATIC_SPAN_TAGS,
+              STATIC_FIELD_TAGS_WITHOUT_ERROR_FIELDS
+            )
+          : orgHasPerformanceView
+            ? Object.assign(
+                {},
+                measurementsWithKind,
+                functionTags,
+                STATIC_SPAN_TAGS,
+                STATIC_FIELD_TAGS
+              )
+            : Object.assign({}, STATIC_FIELD_TAGS_WITHOUT_TRACING);
+
+    Object.assign(combinedTags, filteredTags, STATIC_SEMVER_TAGS, featureFlagTags);
+
+    combinedTags.has = getHasTag(combinedTags);
+
+    return combinedTags;
```

Comment:
I was gonna say this is a bit hard to read but it looks like it's existing code.